test/core/tsi/alts/handshaker/alts_handshaker_client_test.cc

```diff
@@ -228,8 +229,9 @@ static grpc_call_error check_server_start_success(grpc_call* /*call*/,
     return GRPC_CALL_OK;
   }
   upb::Arena arena;
-  alts_handshaker_client* client =
-      static_cast<alts_handshaker_client*>(closure->cb_arg);
+  AltsHandshakerClient* client =
```

Comment:
Throughout, can we refactor to using smart pointers (`std::unique_ptr<AltsHandshakerClient>`) instead of raw pointers?